Add support for serverless v4

## Expected Behavior
`npm i serverless-plugin-datadog --save-dev` should work when using serverless framework version 4+

## Actual Behavior
npm ERR! Could not resolve dependency:
npm ERR! peer serverless@"3.x || 2.x || 1.x" from serverless-plugin-datadog@5.66.0

## Steps to Reproduce the Problem

  1. Install serverless v4
  1. `npm i serverless-plugin-datadog --save-dev`
 
## Specifications

  - Serverless Framework version:  4.1.5
  - Datadog Serverless Plugin version:  5.66.0
  - Lambda function runtime (Python 3.7, Node 10, etc.):  Node v18.16.0

--- a/package.json
+++ b/package.json
@@ -62,6 +62,6 @@
     "simple-git": "^3.16.0"
   },
   "peerDependencies": {
-    "serverless": "3.x || 2.x || 1.x"
+    "serverless": "4.x || 3.x || 2.x || 1.x"
   }
 }

